src/vfs2.c

```diff
@@ -1910,3 +1937,15 @@ void vfs2_ut_sm_relate(sqlite3_file *orig, sqlite3_file *targ)
 	PRE(ftarg->flags & SQLITE_OPEN_MAIN_DB);
 	sm_relate(&forig->entry->wtx_sm, &ftarg->entry->wtx_sm);
 }
+
+void vfs2_ut_make_wal_hdr(uint8_t *buf,
```

Comment:
Yeah, I don't love this. Do you think it would be better to duplicate the WAL header construction code in test_vfs2.c? I could go either way personally.